### PR TITLE
add option for user to indicate SCC structure when using optimization…

### PR DESCRIPTION
**Pull request by sajadabvi**
add option for user to indicate SCC structure when using optimization mode

### CODE REVIEW

**Review by Girish-Anadv-07 — APPROVED** [review]

File: gunfolds/conversions.py [hunk 1]
```diff
@@ -823,7 +826,10 @@ def encode_list_sccs(glist):
     first_graph = True
     for i, g in enumerate(glist):
         if first_graph:
-            SCCS = [s for s in strongly_connected_components(graph2nx(g))]
+            if scc_members:
```
Comment (sergeyplis) [inline gunfolds/conversions.py:829]: This comparison should be 
```suggestion
            if scc_members is not None:
```
see [here](https://peps.python.org/pep-0008/#programming-recommendations)

File: gunfolds/scripts/d_rasl.py
Comment (sergeyplis) [inline gunfolds/scripts/d_rasl.py:1]: I do not think our internal running scripts should be a part of a pip installable package. There should be another repository for our experiments that is on github in neuroneural (possibly private) that has gunfolds as a requirement and where you conduct your research.

File: gunfolds/solvers/clingo_rasl.py [hunk 2]
```diff
@@ -273,7 +276,10 @@ def drasl_command(g_list, max_urate=0, weighted=False, scc=False, dm=None, bdm=N
     n = len(g_list)
     command = clingo_preamble(g_list[0])
     if scc:
-        command += encode_list_sccs(g_list)
+        if weighted and scc_members:
```
Comment (sergeyplis) [inline gunfolds/solvers/clingo_rasl.py:279]: I suggest removing the `if then else` entirely and only leaving one line 
```suggestion
        command += encode_list_sccs(g_list, scc_members)
```
This gives us a chance to specify SCC composition even if we are not doing optimization. Useful for figuring out how much does wrong SCC encoding affect the results.

**Review by sergeyplis — CHANGES_REQUESTED** [review]
I request changes in all files and removal of the script file from the repo.

File: gunfolds/solvers/clingo_rasl.py [hunk 3]
```diff
@@ -273,7 +276,10 @@ def drasl_command(g_list, max_urate=0, weighted=False, scc=False, dm=None, bdm=N
...
+            command += encode_list_sccs(g_list, scc_members)
```
Comment (sergeyplis) [inline gunfolds/solvers/clingo_rasl.py:280]:
```suggestion
```

File: gunfolds/solvers/clingo_rasl.py [hunk 4]
```diff
@@ -273,7 +276,10 @@ def drasl_command(g_list, max_urate=0, weighted=False, scc=False, dm=None, bdm=N
...
+        else:
```
Comment (sergeyplis) [inline gunfolds/solvers/clingo_rasl.py:281]:
```suggestion
```

File: gunfolds/solvers/clingo_rasl.py [hunk 5]
```diff
@@ -273,7 +276,10 @@ def drasl_command(g_list, max_urate=0, weighted=False, scc=False, dm=None, bdm=N
...
+            command += encode_list_sccs(g_list)
```
Comment (sergeyplis) [inline gunfolds/solvers/clingo_rasl.py:282]:
```suggestion
```

Comment (sajadabvi) [reply gunfolds/solvers/clingo_rasl.py]: changes made

File: gunfolds/conversions.py (hunk 1 above)
Comment (sajadabvi) [reply gunfolds/conversions.py]: change made and pushed

File: gunfolds/scripts/d_rasl.py
Comment (sajadabvi) [reply gunfolds/scripts/d_rasl.py:1]: Deleted scripts I have added. I will add the to my personal repository

**Pull request merged**